tests/test_annubes_env.py

```diff
@@ -0,0 +1,170 @@
+    assert custom_env.observation_space.name == {"fixation": 0, "start": 1, "v": 2}
+
+
+def test_action_space(default_env, custom_env):
+    """Test the action space of both default and custom environments.
+
+    This test checks:
+    1. The number of possible actions
+    2. The names and values assigned to each action
+    """
+    assert default_env.action_space.n == 2
+    assert custom_env.action_space.n == 3
+
+    assert default_env.action_space.name == {"fixation": 0, "choice": [0, 1]}
+    assert custom_env.action_space.name == {"fixation": 0, "choice": [0, 0.5, 1]}
+
+
+@pytest.mark.parametrize("env", ["default_env", "custom_env"])
+def test_step(request, env):
+    """Test the step function of the environment.
+
+    This test checks:
+    1. Correct and incorrect actions during fixation period
+    2. Correct and incorrect actions during stimulus period
+    3. Rewards given for different actions
+    4. Termination conditions
+    """
+    # Get the environment fixture
+    env = request.getfixturevalue(env)
+    env.reset()
+
+    # Test fixation period
+    _, reward, terminated, truncated, _ = env.step(0)  # Correct fixation
+    assert not terminated
+    assert not truncated
+    assert reward == 0
+
+    _, reward, terminated, truncated, _ = env.step(1)  # Incorrect fixation
+    assert not terminated
+    assert not truncated
+    assert reward == env.rewards["abort"]
+
+    # Move to stimulus period
+    while env.in_period("fixation"):
+        env.step(0)
+
+    # Test stimulus period
+    _, reward, terminated, truncated, _ = env.step(env.gt_now)  # Correct choice
+    assert not terminated
+    assert not truncated
+    assert reward == env.rewards["correct"]
+
+    env.reset()
+    while env.in_period("fixation"):
+        env.step(0)
+
+    _, reward, terminated, truncated, _ = env.step(
+        (env.gt_now + 1) % env.action_space.n,
+    )  # Incorrect choice
+    assert not terminated
+    assert not truncated
+    assert reward == env.rewards["fail"]
+
+
+def test_initial_state_and_first_reset(default_env):
+    """Test the initial state of the environment and its state after the first reset.
+
+    This test checks:
+    1. Initial values of time step and trial number
+    2. Values of time step and trial number after first reset
+    3. Shape of the first observation
+    4. Presence of trial information after reset
+    """
+    # Check initial state
+    assert default_env.t == 0, "t should be 0 initially"
+    assert default_env.num_tr == 0, "num_tr should be 0 initially"
+
+    # Check state after first reset
+    ob, _ = default_env.reset()
+    assert default_env.t == 100, "t should be 100 after first reset"
+    assert default_env.num_tr == 1, "num_tr should be 1 after first reset"
+    assert isinstance(default_env.trial, dict)
+    assert ob.shape == default_env.observation_space.shape, "Observation shape should match the observation space"
```

Comment:
I think it would be useful to also display the value of the parameter found (this suggestion should do that).

```suggestion
    assert default_env.t == 0, f"{t=}, should be 0 initially"
    assert default_env.num_tr == 0, "f"{num_tr=}, should be 0 initially"

    # Check state after first reset
    ob, _ = default_env.reset()
    assert default_env.t == 100, "f"{t=}, should be 100 after first reset"
    assert default_env.num_tr == 1, "f"{num_tr=}, should be 1 after first reset"
    assert isinstance(default_env.trial, dict)
    assert ob.shape == default_env.observation_space.shape, "f"{observation_space.shape=}, should match the observation space"
```